datastore/longhorn.go

```diff
@@ -383,7 +377,7 @@ func (s *DataStore) ValidateV2DataEngine(v2DataEngineEnabled bool) error {
 		return errors.Wrapf(err, "failed to check volume detachment for %v setting update", types.SettingNameV2DataEngine)
 	}
 	if !volumesDetached {
-		return errors.Errorf("cannot apply %v setting to Longhorn workloads when there are attached volumes", types.SettingNameV2DataEngine)
+		return &types.ErrorNotNow{Reason: fmt.Sprintf("cannot apply %v setting to Longhorn workloads when there are attached volumes", types.SettingNameV2DataEngine)}
```

Comment:
Looks like there is an inconsistency between this case (returned `&types.ErrorNotNow`) and other case (return `errors.Wrapf`). Could you elaborate a bit more?

Reply:
I agree.  The case for setting TaintToleration is similar, for instance.  But it doesn't litter the logs like the two I changed.

Reply:
Made all "not with attached volumes" cases return this specialized error type.